availability-store/src/lib.rs

```diff
@@ -46,57 +73,150 @@ pub struct Config {
 	pub path: PathBuf,
 }
 
+/// Compute gossip topic for the erasure chunk messages given the relay parent,
+/// root and the chunk index.
+///
+/// Since at this point we are not able to use [`network`] directly, but both
+/// of them need to compute these topics, this lives here and not there.
+///
+/// [`network`]: ../polkadot_network/index.html
+pub fn erasure_coding_topic(relay_parent: Hash, erasure_root: Hash, index: u32) -> Hash {
+	let mut v = relay_parent.as_ref().to_vec();
+	v.extend(erasure_root.as_ref());
+	v.extend(&index.to_le_bytes()[..]);
+	v.extend(b"erasure_chunks");
+
+	BlakeTwo256::hash(&v[..])
+}
+
+/// A trait that provides a shim for the [`NetworkService`] trait.
+///
+/// Currently it is not possible to use the networking code in the availability store
+/// core directly due to a number of loop dependencies it require:
```

Comment:
I'm hoping that further Substrate improvements allow us to declare a subprotocol directly in this crate.